Merge missing versions & cleanup

DRSDavidSoft opened the pull request:
Hi there, thanks for maintaining the `ini` file! :D

Since I've merged the changes with the version I'm maintaining, I thought I'd ask you to merge some of the missing sections that exist in the version I already have.

Please review, and if something is useless, let me know so I can delete them.  
There are a couple of alternative values that I've commented out.

Additionally, I've cleaned up a couple of whitespace, and trimmed the values, which will be fixed if this PR is accepted.

## List of changes

### Missing builds
(The '✔️' mark means added, '🚫' means non-existant, empty means it were already in your `ini` file)

| name               | x86 | x64 |
| ------------------ | --- | --- |
| `6.2.9200.22977`   | ✔️   | 🚫  |
| `10.0.10240.18485` | ✔️   | 🚫  |
| `10.0.10240.18818` | ✔️   |     |
| `10.0.14393.3503`  | ✔️   | ✔️   |
| `10.0.14393.3986`  | ✔️   | ✔️   |
| `10.0.14393.4169`  | ✔️   |     |
| `10.0.15063.2283`  | 🚫  | ✔️   |
| `10.0.17763.1369`  | ✔️   |     |
| `10.0.17763.1613`  | ✔️   | ✔️   |
| `10.0.17763.1697`  | ✔️   |     |
| `10.0.18362.657`   | ✔️   |     |
| `10.0.18362.836`   | ✔️   |     |
| `10.0.18362.1316`  | ✔️   |     |
| `10.0.18362.1533`  | ?   | ✔️   |
| `10.0.19041.1`     |     | ✔️   |
| `10.0.19041.662`   | ✔️   |     |
| `10.0.19041.746`   | ✔️   |     |
| `10.0.19041.789`   | ✔️   |     |
| `10.0.19041.964`   | ✔️   |     |

### Alternative values
(These alternative values were added as a commented out line)

| name              | key                    | your value | alternative value |
| ----------------- | ---------------------- | ---------- | ----------------- |
| `10.0.17763.1369` | `SingleUserOffset.x64` | `13469`    | `1339C`           |
| `10.0.17763.1697` | `SingleUserOffset.x64` | `13989`    | `138BC`           |
| `10.0.17763.1971` | `SingleUserOffset.x64` | `0DBFC`    | `0DDC9`           |
| `10.0.18362.1316` | `SingleUserOffset.x64` | `0DD19`    | `0DC4C`           |
| `10.0.19041.662`  | `SingleUserOffset.x64` | `0CAE2`    | `0CA0C`           |
| `10.0.19041.746`  | `SingleUserOffset.x64` | `0CB22`    | `0CA4C`           |
| `10.0.19041.964`  | `SingleUserOffset.x64` | `0CB22`    | `0CA4C`           |
| `10.0.19041.964`  | `SLInitOffset.x64`     | `1AE5C`    | `226B0`           |

### Misc
(No real changes to values, just being a Perfectionist)

- `10.0.17763.292`: added missing comments that were present in my `ini` file
- `10.0.17763.437`: added missing comments that were present in my `ini` file
- `10.0.18362.657`: removed blank line after label
- `10.0.9926.0-SLInit`: removed blank line before label
- `10.0.15063.1746-SLInit`: removed two blank lines before label
- `10.0.16179.1000-SLInit`: moved a blank line separating the x86 and x64 sections
- `10.0.17763.1971-SLInit`: added a blank line separating the x86 and x64 sections
- `10.0.18362.1316-SLInit`: trimmed the leading `0` from the hex values
- `10.0.18362.1533-SLInit`: added a blank line separating the x86 and x64 sections

These issues were mostly present in the original file, so I'd very much appreciate it if you could merge this PR, if there are no side effects in doing so.

Please let me know,
thanks!

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -1844,6 +1883,51 @@ SLInitHook.x64=1
+bMultimonAllowed.x86  =C2FA8
+ulMaxDebugSessions.x86=C2FAC
+bFUSEnabled.x86       =C2FB0
+
```
Why there is an empty line?

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -1922,16 +2032,47 @@ SLInitHook.x64=1
+bMultimonAllowed.x86 =C2FA8
+ulMaxDebugSessions.x86=C2FAC
+bFUSEnabled.x86 =C2FB0
+
```
again empty

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -3889,30 +4050,43 @@ SLInitOffset.x64=1ABFC
 SLInitHook.x64=1
 SLInitOffset.x64=1ACDC
 SLInitFunc.x64=New_CSLQuery_Initialize
+bInitialized.x64 =ECAB0
```
Here should be [10.0.17763.437-SLInit], isn't it?

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -6414,15 +6771,43 @@ bMultimonAllowed.x64  =E8478
+bMultimonAllowed.x86  =C2FA8
+ulMaxDebugSessions.x86=C2FAC
+bFUSEnabled.x86       =C2FB0
+
```
again empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -6414,15 +6771,43 @@ bMultimonAllowed.x64  =E8478
+bMultimonAllowed.x86  =C1FA8
+ulMaxDebugSessions.x86=C1FAC
+bFUSEnabled.x86       =C1FB0
+
```
and again

sebaxakerhtc commented:
So... 
You have to remove empty lines inside section.
Example:
```
[10.0.9999.1000]
line
line

line
```
Should be:
```
[10.0.9999.1000]
line
line
line
```
or 
```
[10.0.9999.1000]
line
line
; x64 section
line
```

sebaxakerhtc commented:
Also take attention at SLInit section

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -1844,6 +1883,51 @@ SLInitHook.x64=1
+SLInitOffset.x64=22C80
+SLInitFunc.x64=New_CSLQuery_Initialize
+
+[10.0.14393.3503-SLInit]
```
SLInit sections should be below in ini file with others

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -1844,6 +1883,51 @@ SLInitHook.x64=1
 SLInitOffset.x64=22C80
 SLInitFunc.x64=New_CSLQuery_Initialize
 
+[10.0.14393.3503]
```
3503 should be AFTER 3383

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -1922,16 +2032,47 @@ SLInitHook.x64=1
 SLInitOffset.x64=22C80
 SLInitFunc.x64=New_CSLQuery_Initialize
 
+[10.0.14393.3986-SLInit]
```
again SLInit should be in second half of file, not right after main section

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -7819,8 +8212,8 @@ bRemoteConnAllowed.x86=CD7AC
 ulMaxDebugSessions.x86=CD7B4
 bFUSEnabled.x86       =CD7B8
-bInitialized.x64      =ECAB0
 
```
Here we need to remove empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -7877,31 +8270,78 @@ bMultimonAllowed.x86  =CD7B4
+bMultimonAllowed.x86  =CD7B4
+ulMaxDebugSessions.x86=CD7B8
+bFUSEnabled.x86       =CD7BC
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -7877,31 +8270,78 @@ bMultimonAllowed.x86  =CD7B4
+bMultimonAllowed.x86  =D3840
+ulMaxDebugSessions.x86=D3844
+bFUSEnabled.x86       =D3848
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -7877,31 +8270,78 @@ bMultimonAllowed.x86  =CD7B4
+bMultimonAllowed.x86  =D1840
+ulMaxDebugSessions.x86=D1844
+bFUSEnabled.x86       =D1848
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -7877,31 +8270,78 @@ bMultimonAllowed.x86  =CD7B4
 bServerSku.x64        =ECAB8
 lMaxUserSessions.x64  =ECABC
 bAppServerAllowed.x64 =ECAC4
+
```
?

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8037,6 +8477,15 @@ ulMaxDebugSessions.x64=F6AA8
+bMultimonAllowed.x86  =D5794
+ulMaxDebugSessions.x86=D5798
+bFUSEnabled.x86       =D579C
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8047,6 +8496,15 @@ ulMaxDebugSessions.x64=F6AA8
+bMultimonAllowed.x86  =D5794
+ulMaxDebugSessions.x86=D5798
+bFUSEnabled.x86       =D579C
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8057,14 +8515,23 @@ ulMaxDebugSessions.x64=F6AA8
+bMultimonAllowed.x86  =D3794
+ulMaxDebugSessions.x86=D3798
+bFUSEnabled.x86       =D379C
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8125,6 +8592,7 @@ bRemoteConnAllowed.x86=CF938
 bMultimonAllowed.x86  =CF93C
 ulMaxDebugSessions.x86=CF940
 bFUSEnabled.x86       =CF944
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8135,6 +8603,15 @@ ulMaxDebugSessions.x64=104018
+bMultimonAllowed.x86  =D1970
+ulMaxDebugSessions.x86=D1974
+bFUSEnabled.x86       =D1978
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8145,6 +8622,15 @@ ulMaxDebugSessions.x64=106048
+bMultimonAllowed.x86  =D0970
+ulMaxDebugSessions.x86=D0974
+bFUSEnabled.x86       =D0978
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8165,6 +8651,15 @@ ulMaxDebugSessions.x64=106048
+bMultimonAllowed.x86  =D0970
+ulMaxDebugSessions.x86=D0974
+bFUSEnabled.x86       =D0978
+
```
empty line

sebaxakerhtc commented on rdpwrap.ini:
```diff
@@ -8185,6 +8680,15 @@ ulMaxDebugSessions.x64=107128
+bMultimonAllowed.x86  =D1A20
+ulMaxDebugSessions.x86=D1A24
+bFUSEnabled.x86       =D1A28
+
```
empty line

DRSDavidSoft commented:
Wow, such obvious mistakes 😅! @sebaxakerhtc I can't believe you actually took the time to review my PR. Thanks!

I've fixed the re-order you [mentioned](https://github.com/sebaxakerhtc/rdpwrap.ini/pull/74#discussion_r776379777), removed the empty lines as per [the correct formatting](https://github.com/sebaxakerhtc/rdpwrap.ini/pull/74#issuecomment-1002643517), and removed a couple of redundant SLIinit sections, and also the [incorrect SLInit definitions in the wrong place](https://github.com/sebaxakerhtc/rdpwrap.ini/pull/74#discussion_r776371807).

Sorry about that! Hopefully it's now in good order.

Additionally, I'm collection information about the values that this PR adds, so that I'd be confident that the offset values are actually correct.  For the alternative iffset values though, I'm sad to say I have no idea how to test and make sure which one is the more correct offset.

Please let me know if any additional changes are required, for now I'm marking the requesters changes that have been changed as done.

DRSDavidSoft commented:
BTW -- Personally I thought that the following format was correct, as many sources (including the original, apparently) used it:

```ini
[10.0.9999.1000]
line.x86
line.x86
line.x86

line.x64
line.x64
...
```

sebaxakerhtc commented:
> including the original, apparently

From the beginning of the original file we can see
```
[6.0.6000.16386]
SingleUserPatch.x86=1
SingleUserOffset.x86=160BF
SingleUserCode.x86=nop
SingleUserPatch.x64=1
SingleUserOffset.x64=65E3E
SingleUserCode.x64=Zero
DefPolicyPatch.x86=1
DefPolicyOffset.x86=15CD8
DefPolicyCode.x86=CDefPolicy_Query_edx_ecx
DefPolicyPatch.x64=1
DefPolicyOffset.x64=5C88F
DefPolicyCode.x64=CDefPolicy_Query_eax_rcx_jmp
```
So I choose to not place empty lines.

sebaxakerhtc commented:
> I can't believe you actually took the time to review my PR. Thanks!

Good work!
Thank you for your time!
